keanu-project/src/main/java/io/improbable/keanu/network/BayesianNetwork.java

```diff
@@ -52,28 +53,54 @@ public Vertex getVertexByLabel(VertexLabel label) {
         return labelMap;
     }
 
-    public List<Vertex> getLatentAndObservedVertices() {
+    List<? extends Vertex> getAllVertices() {
+        return vertices;
+    }
+
+    private interface VertexFilter {
+        boolean filter(boolean isProbabilistic, boolean isObserved, int depth);
+    }
+
+    private List<Vertex> getFilteredVertexList(VertexFilter filter) {
         return vertices.stream()
-            .filter(v -> v.isProbabilistic() || v.isObserved())
+            .filter(v -> filter.filter(v.isProbabilistic(), v.isObserved(), v.getId().getDepth()))
             .collect(Collectors.toList());
     }
 
+    /**
+     * @return The union of getLatentVertices and getObservedVertices
+     */
+    public List<Vertex> getLatentAndObservedVertices() {
+        return getLatentAndObservedVerticesAtDepth(Integer.MAX_VALUE);
+    }
+
+    public List<Vertex> getLatentAndObservedVerticesAtDepth(int maxDepth) {
+        return getFilteredVertexList((isProbabilistic, isObserved, depth)
+            -> (isProbabilistic || isObserved) && maxDepth >= depth);
+    }
+
     /**
      * @return All vertices that are latent (i.e. probabilistic non-observed)
```

Comment:
This is a bit confusing, let's match the style of the other comments in this class to keep it consistent: `All vertices are that are latent or observed`. Call the method `getLatentOrObservedVertices()`? And `getLatentOrObservedVerticesAtDepth`